### PR TITLE
fix flaky integration-async tests

### PR DESCRIPTION
**Pull request by MJedr**
ref: #2107

### CODE REVIEW

File: backend/tests/integration-async/records/indexer/test_literature.py [hunk 1]
```diff
@@ -31,12 +30,11 @@ def assert_record():
...
 
-def assert_es_hits_count(expected_hits_count):
+def assert_es_hits_for_record_id_count(recid, expected_hits_count=1):
     def assert_hits():
         current_search.flush_and_refresh("records-hep")
-        result = es_search("records-hep")
-        result_total = get_value(result, "hits.total.value")
-        assert expected_hits_count == result_total
+        hits = LiteratureSearch().get_record(str(recid)).execute().hits
+        assert expected_hits_count == len(hits)
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_literature.py:37]: I would just check if the control_number is on elasticsearch

File: backend/tests/integration-async/records/indexer/test_literature.py [hunk 2]
```diff
@@ -31,12 +30,20 @@ def assert_record():
...
+        assert hits
+
+    retry_until_pass(assert_hits, retry_interval=5)
+
+
+def assert_record_not_in_es(recid):
+    def assert_hits():
+        current_search.flush_and_refresh("records-hep")
+        hits = LiteratureSearch().query_from_iq(f"recid:{recid}")
+        assert hits
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_literature.py:46]: how this is checking if the recid is not in the ES?

File: backend/tests/integration-async/records/indexer/test_literature.py [hunk 3]
```diff
@@ -31,12 +30,20 @@ def assert_record():
...
+
+    retry_until_pass(assert_hits, retry_interval=5)
+
+
+def assert_record_not_in_es(recid):
+    def assert_hits():
+        current_search.flush_and_refresh("records-hep")
+        hits = LiteratureSearch().query_from_iq(f"recid:{recid}").execute().hits
+        assert not hits
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_literature.py:46]: this will always be `false` no? What is the structure of `hits`?

Comment (MJedr) [reply backend/tests/integration-async/records/indexer/test_literature.py:46]: It's a list of `hits` so if es doesn't find anything matching query it's empty

Comment (MJedr) [reply backend/tests/integration-async/records/indexer/test_literature.py:46]: And I think it should always be empty considering that recid is unique

File: backend/tests/integration-async/records/indexer/test_literature.py [hunk 4]
```diff
@@ -270,7 +277,17 @@ def test_record_created_through_api_is_indexed(inspire_app, clean_celery_session
...
-    assert_es_hits_count(1)
+
+    def assert_record_in_es():
+        results = (
+            LiteratureSearch()
+            .query_from_iq(f"recid:{data['control_number']}")
+            .execute()
+            .hits
+        )
+        assert len(results) == 1
```
Comment (drjova) [inline backend/tests/integration-async/records/indexer/test_literature.py:288]: I would just assert if results

Comment (drjova) [reply backend/tests/integration-async/records/indexer/test_literature.py]: or can't we used the function you have created instead of duplicating it?

Comment (MJedr) [reply backend/tests/integration-async/records/indexer/test_literature.py]: yes, i've just done it

**Review by drjova — APPROVED** [review]

**Pull request merged**